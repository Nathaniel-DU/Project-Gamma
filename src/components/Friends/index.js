import React, { Component } from "react";
import friendsList from '../../utils/friendsList';
import SingleFriend from '../SingleFriend';

export default class Friends extends Component {
    state = {
        friends: [],
        intervalId: ''
    }

    componentDidMount() {
        friendsList()
            .then(data => this.setState({ friends: data }))
            .catch(err => console.log(err));
        const intervalId = setInterval(this.updateFriendsList, 1000*10);
        this.setState({intervalId: intervalId});
    }

    componentWillUnmount() {
        clearInterval(this.state.intervalId);
    }

    updateFriendsList = () => {
        friendsList()
            .then(data => this.setState({ friends: data }))
            .catch(err => console.log(err));
    }

    render() {
        if(this.state.friends.length > 0){
            return (
                <div>
                    {this.state.friends.map((friend, i) => <SingleFriend key={i + '-friendlist'} userId={friend.userId} name={friend.name} location={friend.location} />)}
                </div>
            )
        }else{
            return <div></div>
        }
    }
}