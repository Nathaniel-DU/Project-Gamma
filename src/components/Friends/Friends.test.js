import React from 'react';
import { shallow } from 'enzyme';
import Friends from './index';

describe('Friends', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Friends debug />);
        expect(component).toMatchSnapshot();
    })
})