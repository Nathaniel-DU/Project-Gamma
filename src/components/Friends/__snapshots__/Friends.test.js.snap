// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friends should render correctly in "debug" mode 1`] = `<div />`;
